src/main/java/core/basesyntax/SalaryInfo.java

```
@@ -1,7 +1,59 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        StringBuilder result = new StringBuilder("Report for period " + dateFrom + " - " + dateTo);
+
+        if (names != null && data != null && dateFrom != null && dateTo != null) {
+            try {
+                LocalDate startDate = parseDate(dateFrom);
+                LocalDate endDate = parseDate(dateTo);
+
+                for (String name : names) {
+                    int employeeSalary = calculateEmployeeSalary(data, name, startDate, endDate);
+                    result.append(System.lineSeparator()).append(name)
+                            .append(" - ").append(employeeSalary);
+                }
+            } catch (NumberFormatException e) {
+                result.append("\nError parsing number.");
```

it is better to throw the exception